ApplicationLibCode/ProjectDataModel/RimCaseCollection.cpp

```diff
@@ -35,6 +35,19 @@ RimCaseCollection::RimCaseCollection()
     CAF_PDM_InitFieldNoDefault( &reservoirs, "Reservoirs", "Reservoirs ChildArrayField" );
 }
 
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<RimCase*> RimCaseCollection::cases() const
+{
+    std::vector<RimCase*> caseVector;
+    for ( size_t i = 0; i < reservoirs.size(); i++ )
+    {
+        caseVector.push_back( reservoirs[i] );
+    }
+    return caseVector;
+}
+
```

Comment:
I fail to see why this is not just PdmChildArrayField::childrenByType? E.g.` return reservoirs.childrenByType<RimCase>()`?

If that is not possible it should at least be a modern for-loop.. :smile:

Reply:
I need `std::vector<RimCase*>`, but the method `reservoirs.childrenByType()` gives me `std::vector<RimEclipseCase*>` So to avoid include of `RimEclipseCase` at the calling site I do the conversion here.

Reply:
Converted to modern for-loop